source/slang/slang.cpp

```diff
@@ -2464,34 +2484,72 @@ Session* CompileRequestBase::getSession()
 }
 
 static const Slang::Guid IID_ISlangFileSystemExt = SLANG_UUID_ISlangFileSystemExt;
+static const Slang::Guid IID_SlangCacheFileSystem = SLANG_UUID_CacheFileSystem;
 
 void Linkage::setFileSystem(ISlangFileSystem* inFileSystem)
 {
     // Set the fileSystem
     fileSystem = inFileSystem;
 
+    // Release what's there
+    fileSystemExt.setNull();
+    cacheFileSystem.setNull();
+
     // If nullptr passed in set up default 
     if (inFileSystem == nullptr)
     {
-        fileSystemExt = new Slang::CacheFileSystem(Slang::OSFileSystemExt::getSingleton());
+        cacheFileSystem = new Slang::CacheFileSystem(Slang::OSFileSystemExt::getSingleton());
+        fileSystemExt = cacheFileSystem;
     }
     else
     {
-        // See if we have the full ISlangFileSystemExt interface, if we do just use it
-        inFileSystem->queryInterface(IID_ISlangFileSystemExt, (void**)fileSystemExt.writeRef());
-
-        // If not wrap with CacheFileSystem that emulates ISlangFileSystemExt from the ISlangFileSystem interface
-        if (!fileSystemExt)
+        CacheFileSystem* cacheFileSystemPtr = nullptr;   
+        inFileSystem->queryInterface(IID_SlangCacheFileSystem, (void**)&cacheFileSystemPtr);
+        if (cacheFileSystemPtr)
+        {
+            cacheFileSystem = cacheFileSystemPtr;
+            fileSystemExt = cacheFileSystemPtr;
+        }
+        else 
         {
-            // Construct a wrapper to emulate the extended interface behavior
-            fileSystemExt = new Slang::CacheFileSystem(fileSystem);
+            if (m_requireCacheFileSystem)
```

Comment:
I failed to notice when/where `m_requireCacheFileSystem` gets set. I'm assuming it is when the user is requesting us to dump a repro file? Is there a reason not to just use *that* as the condition here, rather than have a distinct flag?

Reply:
It's because you might not generate the repro via command line option, but through the API. The API as it stands needs a mechnism to set the flag to say in effect we are going to use this to record a repro. 

As it stands if you use the standard slang setup, or a ISlangFileSystem (not Ext), it will actually capture everything needed anyway. Still for completeness the API should expose enabling this.